homeassistant/components/shelly/coordinator.py

```diff
@@ -645,6 +665,21 @@ async def _async_run_connected_events(self) -> None:
         """
         if not self.sleep_period:
             await self._async_connect_ble_scanner()
+        else:
+            await self._async_setup_outbound_websocket()
+
+    async def _async_setup_outbound_websocket(self) -> None:
+        """Set up outbound websocket if it is not enabled."""
+        config = self.device.config
+        if (
+            (ws_config := config.get("ws"))
+            and not ws_config["server"]
```

Comment:
I think it would be good to also check `ws_config["enable"]` in case the outbound websocket is disabled and enable it

Reply:
Are you thinking something like?
```
diff --git a/homeassistant/components/shelly/coordinator.py b/homeassistant/components/shelly/coordinator.py
index f74d5781ae6..03dcdedbb6f 100644
--- a/homeassistant/components/shelly/coordinator.py
+++ b/homeassistant/components/shelly/coordinator.py
@@ -673,7 +673,7 @@ class ShellyRpcCoordinator(ShellyCoordinatorBase[RpcDevice]):
         config = self.device.config
         if (
             (ws_config := config.get("ws"))
-            and not ws_config["server"]
+            and (not ws_config["server"] or not ws_config["enable"])
             and (ws_url := get_rpc_ws_url(self.hass))
         ):
             LOGGER.debug(
```

I'm a bit conflicted on that as I think its ok to set it up if its un-configured, but if they explicitly go in an turn if off it seems like we might not want to override their preferences.

Reply:
While I agree there is something in your claim, I think we reach here only if it is a sleepy device, added to HA and has no entities, in this case we probably need to do our best to help the user, so I would prefer we also enable the WebSocket if it is disabled (as your diff above). 

The case that makes most sense not to reconfigure is if it is enabled and set to another value, since there might be a case that the user set another address due to a proxy or different networking, but I don't see a case for having a sleepy device in HA without WebSocket as it makes it non-functioning.

Reply:
I agree with Shay, when `ws_config` is disabled the device simply doesn't work with Home Assistant, integration in such a case should try to fix the problem.

Reply:
Adjusted in a4d5f727ff97f4ead4e0388c66fd2548de1214a5